streamlit_calendar/frontend/src/components/Calendar.tsx

```diff
@@ -0,0 +1,163 @@
+import timeGridPlugin from "@fullcalendar/timegrid"
+import timelinePlugin from "@fullcalendar/timeline" // premium
+
+import {
+  CalendarOptions,
+  EventApi,
+  EventChangeArg,
+  EventClickArg,
+  EventSourceInput,
+  ViewApi,
+} from "@fullcalendar/core"
+import React from "react"
+import { Streamlit, withStreamlitConnection } from "streamlit-component-lib"
+import { ComponentProps } from "streamlit-component-lib/dist/StreamlitReact"
+import styled from "styled-components"
+import {
+  Callback,
+  DateClickComponentValue,
+  DateClickValue,
+  EventChangeComponentValue,
+  EventChangeValue,
+  EventClickComponentValue,
+  EventClickValue,
+  EventsSetComponentValue,
+  EventsSetValue,
+  ViewValue,
+} from "../types/Calendar.type"
+
+const ENABLED_PLUGINS = [
+  adaptivePlugin,
+  dayGridPlugin,
+  interactionPlugin,
+  listPlugin,
+  multiMonthPlugin,
+  resourceDayGridPlugin,
+  resourceTimeGridPlugin,
+  resourceTimelinePlugin,
+  timeGridPlugin,
+  timelinePlugin,
+]
+
+const FullCalendarWrapper = styled.div<{ $customCSS?: string }>`
+  ${(props) => props.$customCSS}
+`
+
+type Props = ComponentProps<{
+  events?: EventSourceInput
+  options?: CalendarOptions
+  customCSS?: string
+  callbacks?: Callback[]
+  licenseKey?: string
+}>
+
+const CalendarFC: React.FC<Props> = ({
+  args: { events, options, customCSS, callbacks, licenseKey },
+}) => {
+  const getViewValue = (view: ViewApi): ViewValue => ({
+    type: view.type,
+    title: view.title,
+    activeStart: view.activeStart.toISOString(),
+    activeEnd: view.activeEnd.toISOString(),
+    currentStart: view.currentStart.toISOString(),
+    currentEnd: view.currentEnd.toISOString(),
+  })
+
+  const handleDateClick = (arg: DateClickArg) => {
+    const dateClick: DateClickValue = {
+      allDay: arg.allDay,
+      date: arg.date.toISOString(),
+      view: getViewValue(arg.view),
+      resource: arg.resource?.toJSON(),
+    }
+
+    const componentValue: DateClickComponentValue = {
+      callback: "dateClick",
+      dateClick,
+    }
+
+    Streamlit.setComponentValue(componentValue)
+  }
+
+  const handleEventClick = (arg: EventClickArg) => {
+    const eventClick: EventClickValue = {
+      event: arg.event.toJSON(),
+      view: getViewValue(arg.view),
+    }
+
+    const componentValue: EventClickComponentValue = {
+      callback: "eventClick",
+      eventClick,
+    }
+
+    Streamlit.setComponentValue(componentValue)
+  }
+
+  const handleEventChange = (arg: EventChangeArg) => {
+    const eventChange: EventChangeValue = {
+      oldEvent: arg.oldEvent.toJSON(),
+      event: arg.event.toJSON(),
+      relatedEvents: arg.relatedEvents.map((related) => related.toJSON()),
+    }
+
+    const componentValue: EventChangeComponentValue = {
+      callback: "eventChange",
+      eventChange,
+    }
+
+    Streamlit.setComponentValue(componentValue)
+  }
+
+  const handleEventsSet = (events: EventApi[]) => {
```

Comment:
Make sure the `handleEventsSet` handler includes the resource id back to avoid loss of data (ref: https://github.com/im-perativa/streamlit-calendar/commit/7bb8cc43d48e894a44b77f414c21e6c889d49af1)

Reply:
Oh ok, that's fair. Sorry again! I probably overlooked this when modifying the callback. Please also take a look at my fix: 360dadb